src/chrome/content/code/HTTPSRules.js

```diff
@@ -487,44 +535,55 @@ const HTTPSRules = {
     if (!alist) this.log(DBUG, "No applicable list rewriting " + input_uri.spec);
     this.log(DBUG, "Processing " + input_uri.spec);
 
-    var uri = this.sanitiseURI(input_uri);
-
     // Get the list of rulesets that target this host
     try {
-      var rs = this.potentiallyApplicableRulesets(uri.host);
+      var host = input_uri.host
     } catch(e) {
-      this.log(NOTE, 'Could not check applicable rules for '+uri.spec + '\n'+e);
-      return null;
-    }
-
-    // ponder each potentially applicable ruleset, working out if it applies
-    // and recording it as active/inactive/moot/breaking in the applicable list
-    for (i = 0; i < rs.length; ++i) {
-      if (!rs[i].active) {
-        if (alist && rs[i].wouldMatch(uri, alist))
-          alist.inactive_rule(rs[i]);
-        continue;
-      } 
-      blob.newuri = rs[i].transformURI(uri);
-      if (blob.newuri) {
-        if (alist) {
-          if (uri.spec in https_everywhere_blacklist) 
-            alist.breaking_rule(rs[i]);
-          else 
-            alist.active_rule(rs[i]);
-  }
-        if (userpass_present) blob.newuri.userPass = input_uri.userPass;
-        blob.applied_ruleset = rs[i];
-        return blob;
+      // NS_ERROR_FAILURE is normal for accessing uri.host. It just means that
+      // host is not applicable for the URI scheme, e.g. about: URIs.
+      // If that happens we quietly return null. If another exception happens
+      // we noisily return null.
+      if (e.name != "NS_ERROR_FAILURE") {
+        this.log(WARN, 'Could not get host from ' + input_uri.spec + ': ' + e);
       }
-      if (uri.scheme == "https" && alist) {
-        // we didn't rewrite but the rule applies to this domain and the
-        // requests are going over https
-        if (rs[i].wouldMatch(uri, alist)) alist.moot_rule(rs[i]);
-        continue;
-      } 
-    }
-    return null;
+      callback(null);
+      return true;
+    }
+    var that = this;
+    return this.potentiallyApplicableRulesets(host, function(rs) {
```

Comment:
nit: again, give this callback a helpful name rather than defining it inside a call?